sdk/python/lib/pulumi/x/automation/_stack.py

```diff
@@ -620,3 +670,45 @@ def fully_qualified_stack_name(org: str, project: str, stack: str) -> str:
     :returns: The fully qualified stack name.
     """
     return f"{org}/{project}/{stack}"
+
+
+def _create_log_file(command: str) -> Tuple[str, tempfile.TemporaryDirectory]:
+    log_dir = tempfile.TemporaryDirectory(prefix=f"automation-logs-{command}-")
+    filepath = os.path.join(log_dir.name, "eventlog.txt")
+
+    # Open and close the file to ensure it exists before we start polling for logs
+    f = open(filepath, "w+")
+    f.close()
+    return filepath, log_dir
+
+
+def _watch_logs(filename: str, callback: OnEvent):
```

Comment:
I tested this a bit it seems to work pretty well. sleep() does not hold GIL so other threads can make progress. The `f.readline()` keeps returning empty str even when it reached eof. 

Something you might want to think about is what bad things happen when we run user callbacks on the logging thread. It does not seem to be the user's expectation. These callbacks cannot run simultaneously with user Py code because Py code always takes the GIL, but if they utilize sys calls concurrency is possible. I can't construct a confusing example but I sense there might be one here. Let me look up python threading modules to see what else we can do to push the events back to the parent thread and execute the callbacks there.